source/rpt/rpt_cell_reconstruction.cc

```diff
@@ -0,0 +1,626 @@
+{
+  std::ofstream myfile;
+  std::string   sep;
+  std::string   filename =
+    parameters.reconstruction_param.reconstruction_positions_file;
+  myfile.open(filename);
+
+  // Headers with space or comma
+  if (filename.substr(filename.find_last_of(".") + 1) == ".dat")
+    {
+      myfile
+        << "unknown_positions_id volumes particle_positions_x particle_positions_y particle_positions_z cell_level";
+      sep = " ";
+    }
+  else // .csv is default
+    {
+      myfile
+        << "unknown_positions_id,volumes,particle_positions_x,particle_positions_y,particle_positions_z,cell_level";
+      sep = ",";
+    }
+
+  // Add status columns and distance if analyse positions is enable
+  if (parameters.reconstruction_param.analyse_positions)
+    myfile << sep << "status" << sep << "status_id" << sep << "distance"
+           << std::endl;
+  else
+    myfile << std::endl;
+
+  for (unsigned int i_particle = 0;
+       i_particle < reconstruction_positions.size();
+       i_particle++)
+    {
+      myfile << i_particle << sep << cells_volumes[i_particle] << sep
+             << reconstruction_positions[i_particle][0] << sep
+             << reconstruction_positions[i_particle][1] << sep
+             << reconstruction_positions[i_particle][2] << sep
+             << final_cell_level[i_particle];
+
+      if (parameters.reconstruction_param.analyse_positions)
+        {
+          // Calculate distance between known and found positions
+          double distance =
+            std::sqrt(std::pow(reconstruction_positions[i_particle][0] -
+                                 known_positions[i_particle][0],
+                               2) +
+                      std::pow(reconstruction_positions[i_particle][1] -
+                                 known_positions[i_particle][1],
+                               2) +
+                      std::pow(reconstruction_positions[i_particle][2] -
+                                 known_positions[i_particle][2],
+                               2));
+
+          // Associate status to a status id
+          unsigned int status_id;
+          if (cell_status[i_particle] == "right_cell")
+            status_id = 0;
+          else if (cell_status[i_particle] == "cost_function|right_cell")
+            status_id = 1;
+          else if (cell_status[i_particle] == "parent_cell|right_cell")
+            status_id = 2;
+          else if (cell_status[i_particle] ==
+                   "parent_cell|cost_function|right_cell")
+            status_id = 3;
+          else if (cell_status[i_particle] == "wrong_cell")
+            status_id = 4;
+          else if (cell_status[i_particle] == "cost_function|wrong_cell")
+            status_id = 5;
+          else if (cell_status[i_particle] == "parent_cell|wrong_cell")
+            status_id = 6;
+          else if (cell_status[i_particle] ==
+                   "parent_cell|cost_function|wrong_cell")
+            status_id = 7;
```

Comment:
Maybe use a switch case instead of bunch of ifs? Very interesting way of setting status though. You could have used a mask instead of a long string (the ifs would have been faster), but I don't think that's a big deal.

Reply:
Yes it's better to change that. It's funny because I actually implemented the strings and then the status. 
By "mask", do you mean a id number associate to each case that can be processed with
 switch case or determined the status directly in the function?

Reply:
Well a mask is generally a lot of booleans in a single variable. For example 0000 means something false false false false, 0001 means false false false true, etc. This is a good way when you have multiple independent bools. I don't think this is required here.